### PR TITLE
More fixes for patch 2.02

### PR DESCRIPTION
**Pull request by psiberx**

### CODE REVIEW

File: include/RED4ext/RTTISystem-inl.hpp [hunk 1]
```diff
@@ -20,12 +20,17 @@ RED4EXT_INLINE void RED4ext::CRTTISystem::RegisterType(CBaseRTTIType* aType)
...
-    static RTTIRegistrator instance;
+    RED4EXT_UNUSED_PARAMETER(aUnused);
 
-    using func_t = RTTIRegistrator* (*)(RTTIRegistrator*, CallbackFunc, CallbackFunc, bool);
+    if (aRegFunc)
+    {
+        CRTTISystem::Get()->AddRegisterCallback(aRegFunc);
+    }
 
-    RelocFunc<func_t> func(Addresses::CRTTIRegistrator_Add);
```
Comment (WopsS) [inline include/RED4ext/RTTISystem-inl.hpp:27]: I think it is a good idea to remove this address.

File: include/RED4ext/RTTISystem-inl.hpp [hunk 2]
```diff
@@ -20,12 +20,17 @@ RED4EXT_INLINE void RED4ext::CRTTISystem::RegisterType(CBaseRTTIType* aType)
 
 RED4EXT_INLINE void RED4ext::RTTIRegistrator::Add(CallbackFunc aRegFunc, CallbackFunc aPostRegFunc, bool aUnused)
```
Comment (WopsS) [inline include/RED4ext/RTTISystem-inl.hpp:21]: Should we deprecate this?

Comment (psiberx) [reply include/RED4ext/RTTISystem-inl.hpp:21]: Yeah. `[[deprecated("Use 'AddRegisterCallback()' and `AddPostRegisterCallback()` instead.")]]` or something more verbose?

Comment (WopsS) [reply include/RED4ext/RTTISystem-inl.hpp:21]: That is perfect.

**Review by WopsS — APPROVED** [review]
Thanks!

**Pull request merged**